catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,225 @@
+                if (id != null && id.equals(CRON_TIME_ID_KEY)) {
+                    return (String) preference.get(CRON_TIME_KEY + TXT_PREFIX);
+                }
+            }
+        } catch (PersistenceException e) {
+            readWriteLock.readLock()
+                    .unlock();
+            return DEFAULT_CRON_TIME;
+        }
+        return DEFAULT_CRON_TIME;
+    }
+
+    private void setPersistentCronTime() {
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addIdProperty(CRON_TIME_ID_KEY);
+        persistentItem.addProperty(CRON_TIME_KEY, this.cronTime);
+
+        try {
+            readWriteLock.readLock()
+                    .lock();
+            persistentStore.add(PREFERENCES_TYPE, persistentItem);
+
+        } catch (PersistenceException e) {
+            LOGGER.error("Error adding Cron Time to Persistent Store.");
+        } finally {
+            readWriteLock.readLock()
+                    .unlock();
+        }
+    }
+
+    private void registerMbean() {
+        ObjectName objectName = null;
+        MBeanServer mBeanServer = null;
+        try {
+            objectName = new ObjectName(DataUsage.class.getName() + ":service=datausage");
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Data Usage Configuration MBean.", e);
+        }
+        if (mBeanServer != null) {
+            try {
+                try {
+                    mBeanServer.registerMBean(this, objectName);
+                    LOGGER.info("Registered Data Usage Configuration MBean under object name: {}",
+                            objectName.toString());
+                } catch (InstanceAlreadyExistsException e) {
+                    // Try to remove and re-register
+                    mBeanServer.unregisterMBean(objectName);
+                    mBeanServer.registerMBean(this, objectName);
+                    LOGGER.info("Re-registered Data Usage Configuration MBean");
+                }
+            } catch (Exception e) {
+                LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+            }
+        }
+    }
+
+    @Override
+    public Map<String, List<Long>> userMap() {
+        Map<String, List<Long>> dataUsageMap = new HashMap<>();
+        try {
+            List<Map<String, Object>> userMap = attributesStore.getAllUsers();
+            for (Map<String, Object> stringObjectMap : userMap) {
+                if (MapUtils.isNotEmpty(stringObjectMap)) {
+                    dataUsageMap.put((String) stringObjectMap.get(AttributesStore.USER_KEY + TXT_PREFIX),
+                            Arrays.asList((Long) stringObjectMap.get(AttributesStore.DATA_USAGE_KEY + LNG_PREFIX),
+                                    (Long) stringObjectMap.get(AttributesStore.DATA_USAGE_LIMIT_KEY + LNG_PREFIX)));
```

Comment:
This is fine but your friendly java8 reminder could do 
```java
attributeStore.getAllUsers()
    .stream()
    .filter(MapUtils::isNotEmpty)
    .forEach(map -> ...)
```
And I know there's a way that you could reduce straight into a map functionally over side effects but @coyotesqrl is the real pro at that.